src/gentropy/dataset/dataset.py

```diff
@@ -120,10 +144,14 @@ def from_parquet(
             ValueError: Parquet file is empty
         """
         schema = cls.get_schema()
-        df = session.load_data(path, format="parquet", schema=schema, **kwargs)
+
+        # Separate class params from spark params
+        class_params, spark_params = cls._process_class_params(kwargs)
+
+        df = session.load_data(path, format="parquet", schema=schema, **spark_params)
         if df.isEmpty():
             raise ValueError(f"Parquet file is empty: {path}")
-        return cls(_df=df, _schema=schema)
+        return cls(_df=df, _schema=schema, **class_params)
 
```

Comment:
Am I right that this was required to pass the `id_threshold` value to the variant index initialisation?

Reply:
Exactly. I had to distinguish the passed kwargs so that we channeled the spark related ones to load_data, and the class attributes to the instance creation bit.